From 7fbe8ea58189a65d57f342800c695341612fb04d Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Mon, 26 Jan 2026 09:31:46 -0800
Subject: [PATCH 3/4] define extend_hfa_type with inline assembly for clang on
 win arm64

---
 src/aarch64/ffi.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/aarch64/ffi.c b/src/aarch64/ffi.c
index 486fe27..658199a 100644
--- a/src/aarch64/ffi.c
+++ b/src/aarch64/ffi.c
@@ -377,12 +377,15 @@ extend_integer_type (void *source, int type)
     }
 }
 
-#if defined(_MSC_VER)
+#if defined(_MSC_VER) && !defined(__clang__)
 void extend_hfa_type (void *dest, void *src, int h);
 #else
 static void
 extend_hfa_type (void *dest, void *src, int h)
 {
+  #if defined(_MSC_VER)
+  typedef SSIZE_T ssize_t;
+  #endif
   ssize_t f = h - AARCH64_RET_S4;
   void *x0;
 
@@ -448,7 +451,7 @@ extend_hfa_type (void *dest, void *src, int h)
 }
 #endif
 
-#if defined(_MSC_VER)
+#if defined(_MSC_VER) && !defined(__clang__)
 void* compress_hfa_type (void *dest, void *src, int h);
 #else
 static void *
-- 
2.45.2

